### PR TITLE
[Scheduling Networking][2/2] Rest of Matching Networking

### PR DESCRIPTION
**Pull request by pnor**
## Overview

Implement networking to all the matching screens, and the ability to update, and cancel a match

## Changes Made

- Add networking to HomeViewController to get the user and match
- Add networking to the scheduling view controllers to manipulate matches
- Refactor schedulingTimeVC to use DaySchedules instead of strings
- Refactor the scheduling VCs to user the user models for saved availabilities
- Change Home Screen to user the pair to generate info about common talking points
- Add functions/changes to Time+Helper
- Add constants to represent the string days and string match statuses the backend returns
- Combined some models that represented the same times (Schedule and DaySchedule)
- Fix misc bugs in the scheduling view controllers

## Test Coverage

Tested on simulator using local host as the backend
Tested each of the match statuses both by constructing it on frontend and having the backend server return a match

## Related PRs or Issues (delete if not applicable)

#105 #93 #77 

## Screenshots

<details>
  <summary>Doing the match</summary>

normal_small

</details>

<details>
  <summary>Cancelling</summary>

cancel

</details>

### CODE REVIEW

File: CoffeeChat/Controller/HomeViewController.swift [hunk 1]
```diff
@@ -65,42 +65,44 @@ class HomeViewController: UIViewController {
         view.addSubview(tabCollectionView)
 
         setUpConstraints()
+        setUserAndTabPage()
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/HomeViewController.swift:68]: Just a question: do we want networking to be done in viewWillAppear? Asking cause when you just call once in viewDidLoad, if there are any changes and you come back to this screen, the changes won't actually appear.

File: CoffeeChat/Controller/HomeViewController.swift [hunk 2]
```diff
@@ -65,42 +65,44 @@ class HomeViewController: UIViewController {
...
-    private func getUserMatching() {
+    private func setUserAndTabPage() {
+        getUserThen { user in
+            self.user = user
+
+            if let pictureURL = URL(string: user.profilePictureURL) {
+                self.profileButton.kf.setImage(with: pictureURL, for: .normal)
+            }
+
+            let firstActiveMatch = user.matches.filter({ $0.status != "inactive" }).first
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/HomeViewController.swift:79]: Is the backend giving inactive matches too? :o or multiple matches?

File: CoffeeChat/Controller/MatchViewController.swift [hunk 3]
```diff
@@ -8,63 +8,76 @@
 import GoogleSignIn
 import UIKit
 
+/// If the local stored matchID matches the current match from backend, then the user has already reached out
+private func userAlreadyReachedOut(to match: Match) -> Bool {
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/MatchViewController.swift:12]: Note: we may want this to eventually persist in the backend? Realized UserDefaults isn't best place to be storing info about the user, especially the "dynamic" info

Comment (lucyxubroad) [reply CoffeeChat/Controller/MatchViewController.swift:12]: After you read this, can you let me know your thoughts or maybe just make an issue in GitHub if I am not mistaken?

File: CoffeeChat/Controller/MatchViewController.swift [hunk 4]
```diff
@@ -192,37 +246,82 @@ class MatchViewController: UIViewController {
...
+            matchSummaries.append(matchSummary)
+        }
+
+        let exclusivePairInterests = pair.interests.filter { !commonInterests.contains($0) }
+        if exclusivePairInterests.count > 0 {
+            matchSummary = MatchSummary(title: "He also enjoys...", detail: stringListToSentence(words: exclusivePairInterests))
+            matchSummaries.append(matchSummary)
+        }
 
+        let exclusivePairGroups = pair.groups.filter { !commonGroups.contains($0) }
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/MatchViewController.swift:288]: Wondering if there is any way to combine these last four into a single method? But if its too much trouble, probably not a big deal.

File: CoffeeChat/Controller/SchedulingPlacesViewController.swift [hunk 5]
```diff
@@ -37,13 +37,13 @@ class SchedulingPlacesViewController: UIViewController {
...
     private let schedulingStatus: SchedulingStatus
-    private let isChoosing: Bool
+    private var isChoosing: Bool { .choosing ~= schedulingStatus }
     // Location user picked from match's locations
     private var pickedLocation: String?
 
     // Data received from `SchedulingTimeViewController`
-    private var availabilities: [String: [String]] = [:]
-    private var pickedTime: (day: String, time: String) = (day: "", time: "")
+    // private var availabilities: [SubTimeAvailability] = []
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/SchedulingPlacesViewController.swift:45]: Remove this line?

File: CoffeeChat/Models/Match.swift [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+struct Match: Codable {
+
+    let matchID: String
+    let status: String
+    let meetingTime: Float?
+    let users: [String]
+    let availabilities: [DaySchedule]
+
+    /**
```
Comment (lucyxubroad) [inline CoffeeChat/Models/Match.swift:19]: I think we may have decided `///` for formal documentation comments?

File: CoffeeChat/Networking/Endpoints.swift [hunk 7]
```diff
@@ -17,10 +17,12 @@ extension Endpoint {
         #if LOCAL
             Endpoint.config.scheme = "http"
             Endpoint.config.port = 5000
+            Endpoint.config.host = "localhost"
```
Comment (lucyxubroad) [inline CoffeeChat/Networking/Endpoints.swift:20]: Remove this?

**Review by lucyxubroad — APPROVED** [review]

File: CoffeeChat/Controller/MatchViewController.swift (hunk 3 above)
Comment (pnor) [reply CoffeeChat/Controller/MatchViewController.swift:12]: yea, I was talking to Gonzalo about this a couple days ago, but at the moment the matching object doesn't  store who already responded, and who still needs to respond, so I was using user defaults as kind of a hack way to track it. Moving forward though once the backend matching object changes we should def change/delete this

File: CoffeeChat/Controller/HomeViewController.swift (hunk 2 above)
Comment (pnor) [reply CoffeeChat/Controller/HomeViewController.swift]: I was using the user response endpoint to get all the info at once for the user and matches, and there is no endpoint that gets just one match including a user atm

**Pull request merged**